Docs update (#73)
solidity-labs-io/forge-proposal-simulator: docs/actions/print-calldata.md

## Overview

> The following guide explains how to create a GitHub Action that prints the proposal output on PRs. The output includes calldata, newly deployed addresses, changed addresses, and proposal actions. Once the action is executed, the output will be printed in a comment on the PR. It's worth noting that if a PR touches multiple proposals, the comment will only contain the output of the first proposal that the "Get Changed Files" step finds.

Reviewer: ```suggestion
The following guide explains how to create a GitHub Action that prints the proposal output on PRs. The output includes calldata, newly deployed addresses, changed addresses, and proposal actions. Once the action is executed, the output will be printed in a comment on the PR. It's worth noting that if a PR touches multiple proposals, the comment will only contain the output of the proposal with the greatest number.
```

                        return;
                      }
                      for (const comment of actionComments) {
                        await github.rest.issues.deleteComment({
                          ...context.repo,
                          comment_id: comment.id,
                        });
                      }

            - name: Run Proposal
              run: ./run-proposal.sh

            - name: Comment PR with Proposal Output
              uses: actions/github-script@v6
              with:
                  github-token: ${{ secrets.GITHUB_TOKEN }}
                  script: |
                      const fs = require('fs');
                      if (fs.existsSync('output.json')) {
                          const output = JSON.parse(fs.readFileSync('output.json', 'utf8'));
                          const prNumber = context.payload.pull_request.number;
                          github.rest.issues.createComment({
                             ...context.repo,
                             issue_number: prNumber,
                             body: `### Proposal output for ${output.file}:\n\`\`\`\n${output.output}\n\`\`\``
                             });

>                        }

Reviewer 2: this is for another pr but we should just have the .yml files waiting to go in the example repo

Author: you mean we add the github action yml files in example repo and provide their reference here? @ElliotFriedman

Reviewer 2: correct, that is a clean way of doing this

```

To create the workflow, follow these steps:

1. Copy the code above and paste it into a new file named `run-proposal.yml` in the `.github/workflows` folder.
2. Change the `PROPOSALS_FOLDER` to the folder where the proposals are located.
3. Add the Forge Proposal Simulator path before the `run-proposal.sh` script: `lib/forge-proposal-simulator/run-proposal.sh`.

> 4. In case you have proposal names different from `*Proposal_1.sol`, copy the script into your repository and customize it as needed. Update the path in step 3 to `./run-proposal.sh`.

Reviewer: ```suggestion
4. In case you have proposal names different from `*Proposal_1.sol`, copy the [script](https://github.com/solidity-labs-io/fps-example-repo/blob/main/run-proposal.sh) into your repository and customize it as needed. Update the path in step 3 to `./run-proposal.sh`.
```

> 5. Check the repository settings and make sure Read and Write Permissions are enabled in the Workflow Permissions section.

Reviewer 2: ```suggestion
5. Check the Github repository settings and make sure Read and Write Permissions are enabled in the Workflow Permissions section.
```

1. Copy the code above and paste it into a new file named `run-proposal.yml` in the `.github/workflows` folder.
2. Change the `PROPOSALS_FOLDER` to the folder where the proposals are located.
3. Add Forge Proposal Simulator path before the `run-proposal.sh` script: `lib/forge-proposal-simulator/run-proposal.sh`.
4. In case you have proposal names different then `*Proposal_1.sol` copy the script in your repository and customise it as needed. Update the path in step 3 to `./run-proposal.sh`
5. Check the repository settings and make sure Read and Write Permissions
   are enabled on the Worflow Permissions section.

> Whenever a Pull Request that involves a Proposal is created, the action will automatically execute and display the output of the proposal in a comment on the PR. This enables the developer to locally run the proposal and validate whether the output corresponds with the one shown on the PR.x

Reviewer 2: ```suggestion
Whenever a Pull Request that involves a Proposal is created, the action will automatically execute and display the output of the proposal in a comment on the PR. This enables the developer to locally run the proposal and validate whether the output corresponds with the one shown on the PR.
```
